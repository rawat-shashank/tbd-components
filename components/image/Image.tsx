import React from 'react';
import { ImageProps } from './Image.types';
import { StyledImageBase } from './Image.style';

/**
 * Renders an Image component with provided src.
 *
 * >Image component comes with default styles which can be overriden
 * by using either classname or style.
 *
 */
export const Image = ({ src, alt, className, style }: ImageProps) => (
  <StyledImageBase
    src={src}
    alt={alt}
    className={className}
    style={style}
    data-testid="image"
  />
);
