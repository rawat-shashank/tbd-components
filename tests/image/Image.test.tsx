import React from 'react';
import { render } from '@testing-library/react';

import { Default, ImageWithSrc } from '../../stories/image/Image.stories';
import { ThemeProvider } from 'styled-components';

describe('Unit Test | Image Component', () => {
  it('should match default image component', () => {
    const renderComponent = () => render(<Default {...Default.args} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('image');
    expect(testComponent).toHaveStyle({ backgroundColor: '#E7E7E7' });
  });

  it('should match src and alt for ImageWithSrc', () => {
    const renderComponent = () =>
      render(<ImageWithSrc {...ImageWithSrc.args} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('image');
    expect(testComponent).toHaveStyle({ borderRadius: '1.5rem' });
    expect(testComponent).toHaveAttribute('src', ImageWithSrc.args.src);
    expect(testComponent).toHaveAttribute('alt', ImageWithSrc.args.alt);
  });

  it('should override with theme color', () => {
    const theme: any = { elements: { tertiary: 'red' } };
    const renderComponent = () =>
      render(
        <ThemeProvider theme={theme}>
          <Default {...Default.args} />
        </ThemeProvider>
      );
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('image');
    expect(testComponent).toHaveStyle({
      backgroundColor: theme.elements.tertiary,
    });
  });

  it('should match snapshot', () => {
    const { container } = render(<Default {...Default.args} />);
    expect(container).toMatchSnapshot();
  });
});
