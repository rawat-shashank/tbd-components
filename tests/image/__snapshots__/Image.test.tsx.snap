// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test | Image Component should match snapshot 1`] = `
<div>
  <img
    alt=""
    class="sc-bdvvtL fnTSRN"
    data-testid="image"
    src=""
    style="height: 2rem; width: 2rem;"
  />
</div>
`;
